index.js

```diff
@@ -0,0 +1,149 @@
+        fileKeys: []
+      });
+    };
+
+    const loadFiles = (file, isJsonFile, isNotJsonFile) => {
+      // load file if it is a JSON
+      if (file.includes('.json')) {
+        const JSONdata = require(`./locales/${file}`);
+
+        const isDuplicated = new Promise((resolve, reject) => {
+          fs.readFile(`./locales/${file}`, 'utf8', (err, data) => {
+            if (err) {
+              reject(err);
+            } else {
+              const validateDuplicates = jsonValidator.validate(data, false);
+              if (validateDuplicates) {
+                resolve(
+                  `[DUPLICATED] ${jsonValidator.validate(
+                    data,
+                    false
+                  )} => in ${file}`
+                );
+              }
+            }
+          });
+        });
+        isDuplicated.then(res => {
+          console.info(res);
+        });
+        isDuplicated.catch(rej => {
+          console.info(rej);
+        });
+
+        isJsonFile(JSONdata, file);
+      } else {
+        isNotJsonFile;
+      }
+    };
+
+    files.forEach(file => {
+      loadFiles(file, isJsonFile, isNotJsonFile);
+    });
+  };
+  const countKeysBetweenFiles = filesContent => {
+    const singleFinalKeysArr = [];
+    for (const key in filesContent) {
+      const filesContentKeys = filesContent[key]['fileKeys'];
+      for (const k in filesContentKeys) {
+        singleFinalKeysArr.push(filesContentKeys[k]);
+      }
+    }
+    const map = singleFinalKeysArr.reduce((prev, cur) => {
+      prev[cur] = (prev[cur] || 0) + 1;
+      return prev;
+    }, {});
+
+    return map;
+  };
+
+  const countNotJsonFiles = files => {
+    let nonJson = 0;
+    files.forEach(f => {
+      if (!f.includes('.json')) {
+        nonJson++;
+      }
+    });
+    return nonJson;
+  };
+
+  const dspFinalResults = (finalCountedKeys, jsonFiles, filesContent) => {
+    for (const key in finalCountedKeys) {
+      // console.log(notJSON)
+      if (finalCountedKeys[key] < jsonFiles) {
+        const bad_val = `[BAD] ${key}`;
+
+        for (const k in filesContent) {
+          // in case Key not seen
+
+          const filesContentKeys = filesContent[k]['fileKeys'];
+          const fileNameFromK = filesContent[k];
+          const { fileName } = fileNameFromK;
+          if (
+            filesContentKeys.length !== 0 &&
+            !filesContentKeys.includes(key)
+          ) {
+            console.info(`${bad_val} => not seen in ${fileName}`);
+          }
+          for (const n in filesContentKeys) {
+            // throw file name with [BAD]
+            if (
+              key === filesContentKeys[n] &&
+              !filesContentKeys.includes(key)
+            ) {
+              console.info(`${bad_val} => in ${fileName}`);
+            }
+          }
+        }
+      }
+    }
+  };
+
+  fs.readdir(directoryPath, (err, files) => {
+    //handling error
+    if (err) {
+      return console.info('Unable to scan directory: ' + err);
```

Comment:
In that case, is is expected to terminate the program. Please rethrow after logging the error with `throw err`.